rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java

```diff
@@ -0,0 +1,700 @@
+    }
+
+    @Override
+    public <T> ResponseEntity<T> put(String path, Object request, MultiValueMap<String, String> headers, ParameterizedTypeReference<T> responseType) throws RestClientException {
+        ClientResponse response;
+        try {
+            WebClient.RequestBodySpec spec = buildUri(webClient.put(), path)
+                    .headers(h -> {
+                        if (headers != null) {
+                            h.addAll(headers);
+                        }
+                    })
+                    .contentType(config.getContentType())
+                    .accept(config.getAcceptType());
+            response = buildResponseMono(spec, request).block();
+            validateResponse(response, responseType);
+            return response.toEntity(responseType).block();
+        } catch (RestClientException ex) {
+            // Rethrow validation errors
+            throw ex;
+        } catch (Exception ex) {
+            throw new RestClientException("HTTP PUT request failed", ex);
+        }
+    }
+
+    @Override
+    public void putNonBlocking(String path, Object request, Consumer<ClientResponse> onSuccess, Consumer<Throwable> onError) throws RestClientException {
+        putNonBlocking(path, request, null, onSuccess, onError);
+    }
+
+    @Override
+    public void putNonBlocking(String path, Object request, MultiValueMap<String, String> headers, Consumer<ClientResponse> onSuccess, Consumer<Throwable> onError) throws RestClientException {
+        try {
+            WebClient.RequestBodySpec spec = buildUri(webClient.put(), path)
+                    .headers(h -> {
+                        if (headers != null) {
+                            h.addAll(headers);
+                        }
+                    })
+                    .contentType(config.getContentType())
+                    .accept(config.getAcceptType());
+            Mono<ClientResponse> responseMono = buildResponseMono(spec, request);
+            responseMono.subscribe(onSuccess, onError);
+        } catch (Exception ex) {
+            throw new RestClientException("HTTP PUT request failed", ex);
+        }
+    }
+
+    @Override
+    public <T> ObjectResponse<T> putObject(String path, ObjectRequest<?> objectRequest, Class<T> responseType) throws RestClientException {
+        return putObject(path, objectRequest, null ,responseType);
+    }
+
+    @Override
+    public <T> ObjectResponse<T> putObject(String path, ObjectRequest<?> objectRequest, MultiValueMap<String, String> headers, Class<T> responseType) throws RestClientException {
+        ParameterizedTypeReference<ObjectResponse<T>> typeReference = getTypeReference(responseType);
+        ResponseEntity<ObjectResponse<T>> responseEntity = put(path, objectRequest, headers, typeReference);
+        return responseEntity.getBody();
+    }
+
+    @Override
+    public <T> ResponseEntity<T> delete(String path, ParameterizedTypeReference<T> responseType) throws RestClientException {
```

Comment:
HTTP DELETE may contain the request body... 

https://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request

Reply:
WebClient does not have a `body` method for `DELETE`. We could probably work around it, but given that this case is not defined in the RFC, I suggest we avoid using it. Some network components even reject `DELETE` requests with bodies as invalid which could lead to unexpected surprises in production code.